torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -257,6 +257,16 @@ void ProcessGroupAgent::shutdown() {
       recvWork_->abort();
     }
   }
+  // Abort any pending sends to any destination rank.
+  {
+    std::lock_guard<std::mutex> lock(pendingSendMutex_);
+    for (auto& it : currentPendingSends_) {
+      const auto& pendingSends = it.second;
+      for (auto send : pendingSends) {
+        send->abort();
```

Comment:
I'll add a log giving information (sender, receiver) of the send that is being aborted. We should not reach here in a graceful shutdown, since all sends should have been satisfied in a graceful shutdown.